.travis.yml

```diff
@@ -27,26 +27,22 @@ install:
 script: "./build.sh $TEST_SUITE"
 jobs:
   include:
-  - stage: deploy
+  - stage: deploy binaries
     env: GOARCH=amd64 TEST_SUITE=none
-    script: skip
-    deploy:
-    - provider: script
-      skip_cleanup: true
-      script: "./build.sh docker push master"
-      on:
-        branch: master
-    - provider: script
-      skip_cleanup: true
-      script: "./build.sh deploy versioned"
-      on:
-        tags: true
-        branch: master
-    - provider: script
-      skip_cleanup: true
-      script: "./build.sh deploy nightly"
-      on:
-        condition: $TRAVIS_EVENT_TYPE = api
+    script: "./build.sh deploy_binaries"
+    if: tag IS present
+  - stage: deploy packages
```

Comment:
I believe the stage name should be the same for all deployment if you want multiple jobs running in parallel, otherwise it creates a stage for each type of deployment and run each stage sequentially, which significantly increase the total amount time spent for a build:

![screen shot 2018-03-28 at 4 11 21 pm](https://user-images.githubusercontent.com/4105580/38053671-c1cf3246-32a2-11e8-8712-94f627fd09ec.png)

Reply:
Fixed in the latest commit. Thanks!